WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadHandler.java

```diff
@@ -201,25 +205,32 @@ private void finishUpload() {
         private boolean mHasImage, mHasVideo, mHasCategory;
 
         @Override
-        protected void onPostExecute(Boolean pushActionWasDispatched) {
-            if (!pushActionWasDispatched) {
-                // This block only runs if the PUSH_POST action was never dispatched - if it was dispatched, any error
-                // will be handled in OnPostChanged instead of here
-                mPostUploadNotifier.incrementUploadedPostCountFromForegroundNotification(mPost);
-                mPostUploadNotifier.updateNotificationErrorForPost(mPost, mSite, mErrorMessage, 0);
-                finishUpload();
+        protected void onPostExecute(UploadPostTaskResult result) {
+            switch (result) {
+                case ERROR:
+                    mPostUploadNotifier.incrementUploadedPostCountFromForegroundNotification(mPost);
+                    mPostUploadNotifier.updateNotificationErrorForPost(mPost, mSite, mErrorMessage, 0);
+                    finishUpload();
+                    break;
+                case NOTHING_TO_UPLOAD:
```

Comment:
I am guessing the work we do in this `case` is necessary for cleanup. (just trying to make sure I am not missing anything in the review)

Reply:
Yes, the UploadService (even the PostUploadHandler) is basically waiting for the result of the upload -> since we are not uploading anything we need to tell them we are done.